research/models/subject.py

```diff
@@ -143,3 +146,57 @@ def get_raw_information(self) -> pd.Series:
             subject_table["Anonymized", "Patient ID"] == self.id_number
         )
         return subject_table[this_subject]["Raw"].squeeze()
+
+    def get_questionnaire_data(self):
+        """
+        A method to link between a subject to it's questionnaire data.
+
+        Returns
+        -------
+        pd.Series
+            Subject and Questionnaire information.
+        """
+
+        # Getting the raw and anonymized data.
+        anonymized = pd.read_excel(
+            settings.RAW_SUBJECT_TABLE_PATH,
+            sheet_name="Subjects",
+            header=[0, 1],
+            converters={("Raw", "Patient ID"): str},
+            index_col=0,
+        )
+
+        # Getting the questionnaire data from the sheets document.
+        questionnaire = self.subject_data.questionnaire.data
+
+        # Getting the data to intersect subject_id to questionnaire number.
+        decrypted_excel = self.subject_data.open_password_protected_excel(
+            settings.SUBJECT_IDS_TABLE, settings.SUBJECTS_PASSWORD
+        )
+        subjects_id = pd.read_excel(decrypted_excel, converters={"ID": str})
+
+        # Merging tables to get the anonymized_id and questionnaire number.
+        output = anonymized.merge(
+            subjects_id,
+            how="left",
+            left_on=anonymized["Raw"]["Patient ID"],
+            right_on="ID",
+        )
+
+        # Merging tables to get the questionnaire data.
+        output = output.merge(
+            questionnaire,
+            how="left",
+            left_on="Questionnaire",
+            right_on=questionnaire.index,
+        )
```

Comment:
This doesn't belong here. There should be an external function to create the merged dataframe, and this method should simply call it and return the appropriate index.

Reply:
Moved to an external function.